### PR TITLE
disable fetching dependencies if dataspace analysis is found

### PR DESCRIPTION
**Pull request by MauricioUyaguari**
## Summary

 - [x] Disable fetching project dependencies if dataspace cache analysis is found (Improves https://github.com/finos/legend-studio/pull/1509 as fetching dependency is not needed if cache available)
 - [x] Remove usage of `latest` in DeportSeverClient in favor of directly using `MASTER_SNAPSHOT_ALIAS`

## How did you test this change?

- [ ] Test(s) added
- [x] Manual testing (please provide screenshots/recordings)
- [ ] No testing (please provide an explanation)

### CODE REVIEW

File: packages/legend-application-taxonomy/src/stores/StandaloneDataSpaceViewerStore.ts [hunk 1]
```diff
@@ -111,24 +112,27 @@ export class StandaloneDataSpaceViewerStore {
...
         this.graphManagerState.graphManager,
-      ).analyzeDataSpace(dataSpacePath, entities, dependencyEntitiesIndex, () =>
-        retrieveAnalyticsResultCache(
-          project.groupId,
-          project.artifactId,
-          versionId,
-          dataSpacePath,
-          this.depotServerClient,
-        ),
+      ).analyzeDataSpace(
```
Comment (akphi) [inline packages/legend-application-taxonomy/src/stores/StandaloneDataSpaceViewerStore.ts:118]: I think we should move all fetching logic (even fetching entities/project data) in right? We don't need any of them if we just call depot right?

File: .changeset/pretty-feet-double.md [hunk 2]
```diff
@@ -0,0 +1,6 @@
+---
+'@finos/legend-application-taxonomy': patch
+'@finos/legend-extension-dsl-data-space': patch
+---
+
+disable fetching project dependencies if dataspace cache analysis is found
```
Comment (akphi) [inline .changeset/pretty-feet-double.md:6]:
```suggestion
```

**Review by akphi — CHANGES_REQUESTED** [review]

File: packages/legend-application-taxonomy/src/stores/StandaloneDataSpaceViewerStore.ts (hunk 1 above)
Comment (MauricioUyaguari) [reply packages/legend-application-taxonomy/src/stores/StandaloneDataSpaceViewerStore.ts:110]: we should fetch projectData but i agree on the entities

**Review by akphi — CHANGES_REQUESTED** [review]

File: packages/legend-extension-dsl-data-space/src/graphManager/protocol/pure/v1/V1_DSL_DataSpace_PureGraphManagerExtension.ts [hunk 3]
```diff
@@ -86,9 +89,14 @@ export class V1_DSL_DataSpace_PureGraphManagerExtension extends DSL_DataSpace_Pu
...
-    const analysisResult =
-      cachResult ??
-      (await engineClient.postWithTracing<
+    let analysisResult: PlainObject<V1_DataSpaceAnalysisResult>;
+    if (cachResult) {
+      analysisResult = cachResult;
+    } else {
+      actionState?.setMessage('Fetching project entities with dependencies...');
+      const entities = await entitiesRetriever();
+      actionState?.setMessage('Analyzing data space......');
```
Comment (akphi) [inline packages/legend-extension-dsl-data-space/src/graphManager/protocol/pure/v1/V1_DSL_DataSpace_PureGraphManagerExtension.ts:98]: --> `Analyzing data space...`

File: packages/legend-extension-dsl-data-space/src/graphManager/protocol/pure/v1/V1_DSL_DataSpace_PureGraphManagerExtension.ts [hunk 4]
```diff
@@ -69,13 +69,16 @@ export class V1_DSL_DataSpace_PureGraphManagerExtension extends DSL_DataSpace_Pu
...
-    dependencyEntitiesIndex: Map<string, Entity[]>,
+    entitiesRetriever: () => Promise<Entity[]>,
     cacheRetriever?: () => Promise<PlainObject<DataSpaceAnalysisResult>>,
+    actionState?: ActionState,
   ): Promise<DataSpaceAnalysisResult> {
     let cachResult: PlainObject<V1_DataSpaceAnalysisResult> | undefined;
     if (cacheRetriever) {
       try {
+        actionState?.setMessage(
+          'Fetching Cached Data Space Analysis Result...',
```
Comment (akphi) [inline packages/legend-extension-dsl-data-space/src/graphManager/protocol/pure/v1/V1_DSL_DataSpace_PureGraphManagerExtension.ts:80]:
```suggestion
          'Fetching data space analysis result from cache...',
```

File: packages/legend-extension-dsl-data-space/src/graphManager/protocol/pure/v1/V1_DSL_DataSpace_PureGraphManagerExtension.ts [hunk 5]
```diff
@@ -86,9 +89,14 @@ export class V1_DSL_DataSpace_PureGraphManagerExtension extends DSL_DataSpace_Pu
...
     }
     const engineClient = this.graphManager.engine.getEngineServerClient();
-    const analysisResult =
-      cachResult ??
-      (await engineClient.postWithTracing<
+    let analysisResult: PlainObject<V1_DataSpaceAnalysisResult>;
+    if (cachResult) {
+      analysisResult = cachResult;
+    } else {
+      actionState?.setMessage('Fetching project entities with dependencies...');
```
Comment (akphi) [inline packages/legend-extension-dsl-data-space/src/graphManager/protocol/pure/v1/V1_DSL_DataSpace_PureGraphManagerExtension.ts:96]: --> `Fetching project entities and dependencies...`

File: packages/legend-extension-dsl-data-space/src/stores/query/DataSpaceAdvancedSearchState.ts [hunk 6]
```diff
@@ -164,39 +166,25 @@ export class DataSpaceAdvancedSearchState {
...
-      this.loadDataSpaceState.setMessage(`Fetching dependencies...`);
-      const dependencyEntitiesIndex = (yield flowResult(
-        this.depotServerClient.getIndexedDependencyEntities(
-          project,
-          dataSpace.versionId,
-        ),
-      )) as Map<string, Entity[]>;
-
       // analyze data space
-      this.loadDataSpaceState.setMessage(`Analyzing data space...`);
```
[the remark this answers is not in the document]
Comment (MauricioUyaguari) [reply packages/legend-extension-dsl-data-space/src/stores/query/DataSpaceAdvancedSearchState.ts:185]: as discussed moved to the pureGraphManagerPlugin

File: packages/legend-extension-dsl-data-space/src/graphManager/action/analytics/DataSpaceAnalysisHelper.ts [hunk 7]
```diff
@@ -15,24 +15,43 @@
...
+  ProjectData,
+} from '@finos/legend-server-depot';
 import type { PlainObject } from '@finos/legend-shared';
+import type { Entity } from '@finos/legend-storage';
 import type { DataSpaceAnalysisResult } from './DataSpaceAnalysis.js';
 
 const DATASPACE_ANALYTICS_FILE_NAME = 'AnalyticsResult.json';
 const V1_DATASPACE_ANALYTICS_ARTIFACT_EXTENSION_KEY = 'dataSpace-analytics';
 
+export const retrieveProjectEntitiesWithDependencies = async (
```
[the remark this answers is not in the document]
Comment (MauricioUyaguari) [reply packages/legend-extension-dsl-data-space/src/graphManager/action/analytics/DataSpaceAnalysisHelper.ts]: we can't have legend-storage depending on depotServerClient. I moved it to depot package

File: .changeset/fuzzy-trees-suffer.md [hunk 8]
```diff
@@ -0,0 +1,5 @@
+---
+'@finos/legend-server-depot': patch
+---
+
+Remove usage of `latest` in DeportSeverClient in favor of directly using `MASTER_SNAPSHOT_ALIAS`
```
Comment (akphi) [inline .changeset/fuzzy-trees-suffer.md:5]:
```suggestion
Remove usage of `latest` version alias in `DepotServerClient` in favor of directly using `MASTER_SNAPSHOT_ALIAS`
```

**Review by akphi — APPROVED** [review]

**Pull request merged**